src/gui/attributeformconfig/qgsattributewidgetedit.cpp

```diff
@@ -115,6 +116,27 @@ void QgsAttributeWidgetRelationEditWidget::setRelationEditorConfiguration( const
       setCardinalityCombo( QStringLiteral( "%1 (%2)" ).arg( nmrel.referencedLayer()->name(), nmrel.fieldPairs().at( 0 ).referencedField() ), nmrel.id() );
   }
 
+  int widgetTypeIdx = mWidgetTypeComboBox->findData( config.mRelationWidgetType );
+  mWidgetTypeComboBox->setCurrentIndex( widgetTypeIdx >= 0 ? widgetTypeIdx : 0 );
+
+  const QString widgetType = mWidgetTypeComboBox->currentData().toString();
+  mConfigWidget = QgsGui::relationWidgetRegistry()->createConfigWidget( widgetType, relation, this );
+  mConfigWidget->setConfig( config.mRelationWidgetConfig );
+  mWidgetTypePlaceholderLayout->addWidget( mConfigWidget );
+
+  disconnect( mWidgetTypeComboBoxConnection );
+
+  mWidgetTypeComboBoxConnection = connect( mWidgetTypeComboBox, &QComboBox::currentTextChanged, this, [ = ]()
+  {
+    const QString widgetId = mWidgetTypeComboBox->currentData().toString();
+
+    mConfigWidget->deleteLater();
```

Comment:
I didn't see any side effect without removing it from the layout first, but I have added `mWidgetTypePlaceholderLayout->removeWidget( mConfigWidget );` just in case